evmrpc/block.go

```diff
@@ -118,10 +123,13 @@ func (a *BlockAPI) GetBlockTransactionCountByNumber(ctx context.Context, number
 	if err != nil {
 		return nil, err
 	}
-	block, err := blockByNumberWithRetry(ctx, a.tmClient, numberPtr, 1)
+	block, err := blockByNumberRespectingWatermarks(ctx, a.tmClient, a.watermarks, numberPtr, 1)
 	if err != nil {
 		return nil, err
 	}
+	if err := a.ensureHeightAvailable(ctx, block.Block.Height); err != nil {
```

Comment:
Why calling ensureHeightAvailable again here? Shouldn't this be handled already in `blockByNumberRespectingWatermarks` ?